### PR TITLE
docs

### PR DESCRIPTION
**Pull request by manno**
add to the existing docs

### CODE REVIEW

File: docs/about_operators.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# About Operatorso
```
Comment (aduffeck) [inline docs/about_operators.md]:
```suggestion
# About Operators
```

File: docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Testing
+
+Based on upstreams documentation https://github.com/thtanaka/kubernetes/blob/master/docs/devel/testing.md we use three levels of testing: unit, integration and e2e.
+
+Running `make test` executes all the test suites.
+
+We use ginkgo for testing. Every package needs a `suite_test.go` for setup. It can be generated by running `ginkgo bootstrap` in the sub folder. Rename the generated file afterwards.
```
Comment (aduffeck) [inline docs/testing.md]: Maybe explain why the file needs to be renamed? (it's not clear to me right now)

File: docs/testing.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## Unit
+
+While unit testing we:
+
+* test classes in isolation
+* pass all dependencies to the constructor, so we can inject fakes for testing
+* use `counterfeiter` to generate and update fakes
+* don't test private methods, tests are in a separate `_test` package
+* try not to nest ginkgo contexts to deep and keep tests DRY by extracting useful helpers
```
Comment (aduffeck) [inline docs/testing.md]:
```suggestion
* try not to nest ginkgo contexts too deep and keep tests DRY by extracting useful helpers
```

File: docs/testing.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+Running `make test` executes all the test suites.
+
+We use ginkgo for testing. Every package needs a `suite_test.go` for setup. It can be generated by running `ginkgo bootstrap` in the sub folder. Rename the generated file afterwards.
+There is also `ginkgo generate` to create skeleton test files.
+
+## Unit
+
+While unit testing we:
+
+* test classes in isolation
```
Comment (aduffeck) [inline docs/testing.md]:
```suggestion
* test packages in isolation
```
?

File: docs/testing.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+* don't test private methods, tests are in a separate `_test` package
+* try not to nest ginkgo contexts to deep and keep tests DRY by extracting useful helpers
+* assert incoming messages produce the expected state
+* assert outgoing commands happened, like a file gets written
+* assert all handled error cases are triggered
+* can ignore outgoing queries, which only change internal state
+
+## Integration
+
+Integration tests make expectations on the interactions of several components.
```
Comment (aduffeck) [inline docs/testing.md]:
```suggestion
Integration tests formulate expectations on the interactions of several components.
```
Or `validate`? `verify`?

File: docs/testing.md (hunk 2 above)
Comment (manno) [reply docs/testing.md]: We recently renamed all the existing files. Also all controllers are now just named `controller.go`. So it is mostly about consistency.

**Pull request merged**